### PR TITLE
[AL 3826]Show reply message in popup when it is not visible in the list.

### PR DESCRIPTION
**Pull request by Sunilkumarr**
## Summary
Added support for reply message view for older messages 
## Testing

Tested on the device for older messages that message will show  in pop up

### CODE REVIEW

File: Sources/Controllers/ALKReplyController.swift [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+        label.textColor = UIColor(red: 140 / 255.0, green: 140 / 255.0, blue: 140 / 255.0, alpha: 1.0)
+        label.font = Font.light(size: 14.0).font()
+        return label
+    }()
+
+    private lazy var confirmButton: UIButton = {
+        let button = UIButton()
+        let title = localizedString(forKey: "OkayMessage", withDefaultValue: SystemMessage.ButtonName.okay, fileName: configuration.localizedStringFileName)
+        button.setTitle(title, for: .normal)
+        button.setTitleColor(UIView().tintColor, for: .normal)
```
Comment (ShivamPokhriyal) [inline Sources/Controllers/ALKReplyController.swift]: What is UIVIew().tintColor here?

File: Sources/Controllers/ALKReplyController.swift [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+            messageView.isHidden = true
+            attachmentView.update(message: messageObject, localizationFileName: configuration.localizedStringFileName)
+        } else {
+            messageView.isHidden = false
+            attachmentView.isHidden = true
+            modalView.constraint(withIdentifier: ConstraintIdentifier.modelViewHeight)?.constant = Padding.ModelView.heightForMessageView
+            messageView.text = messageObject.message
+            messageView.sizeToFit()
+        }
+        let placeHolder = UIImage(named: "placeholder", in: Bundle.applozic, compatibleWith: nil)
```
Comment (ShivamPokhriyal) [inline Sources/Controllers/ALKReplyController.swift]: Above you've used contactPlaceholder and here just placeholder. I guess contactPlaceholder is the new one, just check which one to use.

File: Sources/Controllers/ALKReplyController.swift [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+        guard let value = message?.createdAtTime.doubleValue else {
+            return
+        }
+
+        let date = Date(timeIntervalSince1970: Double(value / 1000))
+
+        timeLabel.text = date.stringCompareCurrentDate()
+    }
+
+    @objc func tappedCancel() {
```
Comment (ShivamPokhriyal) [inline Sources/Controllers/ALKReplyController.swift]: This is not used I guess, You can remove it!

File: Sources/Controllers/ALKReplyController.swift [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+        modalView.centerXAnchor.constraint(equalTo: view.centerXAnchor).isActive = true
+        modalView.centerYAnchor.constraint(equalTo: view.centerYAnchor).isActive = true
+
+        modalView.heightAnchor.constraintEqualToAnchor(
+            constant: 0,
+            identifier: ConstraintIdentifier.modelViewHeight
+        ).isActive = true
+        modalView.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: Padding.ModelView.left).isActive = true
+        modalView.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -Padding.ModelView.right).isActive = true
+        modalView.addViewsForAutolayout(views: [timeLabel, alertMessageLabel, buttonUIView, confirmButton, avatarImageView, buttonUIView, messageView, attachmentView])
```
Comment (ShivamPokhriyal) [inline Sources/Controllers/ALKReplyController.swift]: buttonUIView is added twice

File: Sources/Controllers/ALKReplyController.swift [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+        avatarImageView.leadingAnchor.constraint(
+            equalTo: modalView.leadingAnchor,
+            constant: Padding.AvatarImage.left
+        ).isActive = true
+        avatarImageView.heightAnchor.constraint(equalToConstant: Padding.AvatarImage.height).isActive = true
+        avatarImageView.widthAnchor.constraint(equalToConstant: Padding.AvatarImage.width).isActive = true
+
+        timeLabel.leadingAnchor.constraint(equalTo: avatarImageView.trailingAnchor, constant: Padding.TitleLabel.left).isActive = true
+        timeLabel.heightAnchor.constraint(equalToConstant: Padding.TitleLabel.height).isActive = true
+        timeLabel.topAnchor.constraint(equalTo: modalView.topAnchor, constant: Padding.TitleLabel.top).isActive = true
```
Comment (ShivamPokhriyal) [inline Sources/Controllers/ALKReplyController.swift]: timeLabel trailingAnchor or widthAnchor is missing. Add any one of these.

File: Sources/Controllers/ALKReplyController.swift [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+
+        messageView.topAnchor.constraint(equalTo: timeLabel.bottomAnchor, constant: -Padding.MessageTextView.top).isActive = true
```
Comment (ShivamPokhriyal) [inline Sources/Controllers/ALKReplyController.swift]: I guess the constant should be +ve here.

File: Sources/Controllers/ALKReplyController.swift [hunk 7]
```diff
@@ -0,0 +1,265 @@
...
+        messageView.leadingAnchor.constraint(equalTo: avatarImageView.trailingAnchor, constant: Padding.MessageTextView.left).isActive = true
+        messageView.trailingAnchor.constraint(equalTo: modalView.trailingAnchor, constant: -Padding.MessageTextView.right).isActive = true
+        messageView.heightAnchor.constraint(equalToConstant: Padding.MessageTextView.height).isActive = true
+
+        attachmentView.topAnchor.constraint(equalTo: timeLabel.bottomAnchor, constant: -Padding.AttachmentView.top).isActive = true
```
Comment (ShivamPokhriyal) [inline Sources/Controllers/ALKReplyController.swift]: Here also, it should be +ve.

File: Sources/Utilities/ALMessage+Extension.swift [hunk 8]
```diff
@@ -218,7 +218,7 @@ extension ALMessage {
 
     var date: Date {
         guard let time = createdAtTime else { return Date() }
-        let sentAt = Date(timeIntervalSince1970: Double(time.doubleValue / 1000))
+        let sentAt = Date(timeIntervalSince1970: Double(createdAtTime.doubleValue / 1000))
```
Comment (ShivamPokhriyal) [inline Sources/Utilities/ALMessage+Extension.swift]: Should be time here not createdAtTime.

File: Sources/Views/ALKAttatchmentView.swift [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+//  Created by apple on 18/10/19.
+//
+
+import Applozic
+import Foundation
+import Kingfisher
+import UIKit
+
+class ALKAttatchmentView: UIView {
+    let loadingIndicator = ALKLoadingIndicator(frame: .zero, color: UIColor.gray)
```
Comment (ShivamPokhriyal) [inline Sources/Views/ALKAttatchmentView.swift]: Nice 🥇

File: Sources/Utilities/ALKFileUtils.swift [hunk 10]
```diff
@@ -68,4 +68,18 @@ class ALKFileUtils: NSObject {
         let fileTypes = ["docx", "pdf", "doc", "java", "js", "txt", "html", "xlsx", "xls", "ppt", "pptx"]
         return fileTypes.contains(pathExtension)
     }
+
+    func getThumbnail(filePath: URL) -> UIImage? {
```
Comment (ShivamPokhriyal) [inline Sources/Utilities/ALKFileUtils.swift]: Nice 👍

File: Sources/Controllers/ALKReplyController.swift (hunk 1 above)
Comment (Sunilkumarr) [reply Sources/Controllers/ALKReplyController.swift]: its default system tint color

File: Sources/Controllers/ALKReplyController.swift (hunk 2 above)
Comment (Sunilkumarr) [reply Sources/Controllers/ALKReplyController.swift]: We use a placeholder in message cell and other places, not the contactPlaceholder

**Pull request merged**